src/System.CommandLine/Command.cs

```diff
@@ -149,6 +149,30 @@ public void SetAction(Func<ParseResult, CancellationToken, Task> action)
             });
         }
 
+        /// <summary>
+        /// Sets an asynchronous action to be run when the command is invoked.
+        /// </summary>
+        /// <remarks>
+        /// When possible, prefer using the <see cref="SetAction(Func{ParseResult, CancellationToken, Task})"/> overload
+        /// and passing the <see cref="CancellationToken"/> parameter to the async method(s) called by the action.
+        /// </remarks>
+        // Hide from intellisense, it's public to avoid the compiler choosing a sync overload
+        // for an async action (and fire and forget issue described in https://github.com/dotnet/command-line-api/issues/2562).
+        [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
If you wanted to force folks to think about the `CancellationToken` you could also mark it obsolete -- not sure how hardcore you are about that.

Reply:
It was my first idea, but then I realized that we should most likely not release new public `[Obsolete]` methods. But I agree, it would be even better as it would allow us emit a compiler warning/error with a clear message.

@jonsequitur thoughts?

Reply:
> Would it be simpler to differentiate the methods by name rather than using overloads, e.g. `SetAction` and `SetAsyncAction`?

I like the `SetAsyncAction` name idea (we never discussed actions at the API review), but users could still run into this particular bug and it would cause another wave of breaking changes in SDK/VMR etc. Let's discuss it offline, I think it's now or never.